lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
Maybe it's worth trying to allocate a big buffer and manually append data to it if there is enough space or if there is not enough space then copy both of them into a new bigger buffer till it's size grows to `MAX_PACKET_SIZE` (same approach may be used in `_parseRemains` to avoid creating a new buffer 2 more times)? Or maybe there were problems with this approach?